numpy/_core/fromnumeric.py

```diff
@@ -49,8 +64,10 @@ def _wrapit(obj, method, *args, **kwds):
 
 
 def _wrapfunc(obj, method, *args, **kwds):
-    bound = getattr(obj, method, None)
-    if bound is None:
+    try:
```

Comment:
Instead of adding an indirection and therefore slowing down the common case where the method is defined correctly, how about going more directly to the goal, and just checking `TypeError` more consistently as one tries calling?  Here, it would seem the code already does the right thing without any changes`, but in other branches, it needs a `try/except` around the trial to call it.

Reply:
My choice was to keep the two logical steps separate:
- first retrive a callable method
- then try if the interface is correct

This is for readability and clarity. I added also the `_get_method` method function to clarly signal that the first step is repeated in different portions of the code.

For the efficiency, please see comment below.